components/elm/src/main/elm_driver.F90

```diff
@@ -286,22 +285,20 @@ subroutine elm_drv(doalb, nextsw_cday, declinp1, declin, rstwr, nlend, rdate)
        call alt_calc(filter(nc)%num_soilc, filter(nc)%soilc, &
             temperature_vars, canopystate_vars) 
 
-       if (use_cn) then
-          !  Note (WJS, 6-12-13): Because of this routine's placement in the driver sequence
-          !  (it is called very early in each timestep, before weights are adjusted and
-          !  filters are updated), it may be necessary for this routine to compute values over
-          !  inactive as well as active points (since some inactive points may soon become
-          !  active) - so that's what is done now. Currently, it seems to be okay to do this,
-          !  because the variables computed here seem to only depend on quantities that are
-          !  valid over inactive as well as active points.
-
-          call decomp_vertprofiles(bounds_clump, &
-               filter_inactive_and_active(nc)%num_soilc, &
-               filter_inactive_and_active(nc)%soilc, &
-               filter_inactive_and_active(nc)%num_soilp, &
-               filter_inactive_and_active(nc)%soilp, &
-               soilstate_vars, canopystate_vars, cnstate_vars)
-       end if
+       !  Note (WJS, 6-12-13): Because of this routine's placement in the driver sequence
```

Comment:
Why delete `if (use_cn)`?

Reply:
one of the objectives of this PR is to make it so that when fates is turned on, as much bgc code that can be shared with CN is used.  But now that I look at it, in this case, the decomp_verprofiles() routine will be called for use_cn and use_fates (which is intended), but it will also potentially be used for use_sp (not intended) which is unecessary. I will add in a logical filter for use_fates and use_cn